source/material_model/interface.cc

```diff
@@ -303,12 +304,28 @@ namespace aspect
       velocity(fe_values.n_quadrature_points, numbers::signaling_nan<Tensor<1,dim> >()),
       composition(fe_values.n_quadrature_points, std::vector<double>(introspection.n_compositional_fields, numbers::signaling_nan<double>())),
       strain_rate(fe_values.n_quadrature_points, numbers::signaling_nan<SymmetricTensor<2,dim> >()),
-      cell(cell_x)
+      cell(cell_x),
+      current_cell(*cell_x)
```

Comment:
What happens if `cell_x` is a `NULL` pointer? Don't you have to do something like
```
  current_cell (cell_x == NULL ? typename DoFHandler<dim>::active_cell_iterator() : *cell_x)
```
?

I think it would probably be good to also do
```
  cell (cell_x == NULL ? NULL : &current_cell)
```
This way, the calling site definitely no longer needs to keep the pointed to object around, and this avoids a whole class of possible errors.